test/jdk/java/lang/StringBuffer/IndexOf.java

```diff
@@ -27,162 +27,396 @@
+  String endNeedle = new String(
+      Arrays.copyOfRange((needleCharset == StandardCharsets.UTF_16) ? haystack_16 : haystack, endStart + l_offset, endStart + needleSize + l_offset));
+  // String shs = new String(Arrays.copyOfRange((hs_charset == StandardCharsets.UTF_16) ? haystack_16 : haystack, 0, haystackSize));
+  String shs = (new String((hs_charset == StandardCharsets.UTF_16) ? haystack_16 : haystack)).substring(0, haystackSize);
+
+  shs = "$&),,18+-!'8)+";
+  endNeedle = "8)-";
+  l_offset = 9;
+  endNeedle = "/'!(\"3//";
+  shs = ");:(/!-+ %*/'!(\"3//;9";
+  l_offset = 0;
+  StringBuffer bshs = new StringBuffer(shs);
+
+  // printStringBytes(shs.getBytes(hs_charset));
+  for (int i = 0; i < 200000; i++) {
+    if(shs.indexOf(endNeedle, l_offset) != -1) {
+      // System.out.println("result="+bshs.indexOf(endNeedle, l_offset));
+    }
+    result += bshs.indexOf(endNeedle, l_offset);
+    // System.out.print(result + " " + needle + " " + shs);
+  }
 
-    public static void main(String[] args) throws Exception {
-        simpleTest();
-        compareIndexOfLastIndexOf();
-        compareStringStringBuffer();
+  ///////////////////////////////////////////////////////////////////////////////////
+  ///////////////////////////////////////////////////////////////////////////////////
 
-        if (failure)
-           throw new RuntimeException("One or more BitSet failures.");
-    }
 
-    private static void report(String testName, int failCount) {
-        System.err.println(testName+": " +
-                         (failCount==0 ? "Passed":"Failed("+failCount+")"));
-        if (failCount > 0)
-            failure = true;
+    for (int i = 0; i < 20000; i++) {
+      int foo = testName.indexOf("dex");
     }
-
-    private static String generateTestString(int min, int max) {
-        StringBuffer aNewString = new StringBuffer(120);
-        int aNewLength = getRandomIndex(min, max);
-        for(int y=0; y<aNewLength; y++) {
-            int achar = generator.nextInt(30)+30;
-            char test = (char)(achar);
-            aNewString.append(test);
-        }
-        return aNewString.toString();
+    System.out.println("");
+    simpleTest();
+    compareIndexOfLastIndexOf();
+    compareStringStringBuffer();
+    compareExhaustive();
+
+    if (failure)
+      throw new RuntimeException("One or more BitSet failures.");
+  }
+
+  private static void report(String testName, int failCount) {
+    System.err.println(testName + ": " +
+        (failCount == 0 ? "Passed" : "Failed(" + failCount + ")"));
+    if (failCount > 0)
+      failure = true;
+  }
+
+  private static String generateTestString(int min, int max) {
+    StringBuffer aNewString = new StringBuffer(120);
+    int aNewLength = getRandomIndex(min, max);
+    for (int y = 0; y < aNewLength; y++) {
+      int achar = generator.nextInt(30) + 30;
```

Comment:
This will only ever generate LL cases, i.e. chars from [30,60]. Could be parametrized to also produce utf16 if instead of 30, offset was in the unicode range